### PR TITLE
 Initialization for intermediate primary with CLONE

### PR DESCRIPTION
**Pull request by kfyharukz**
Signed-off-by: Kazuhito MATSUDA <kazuhito.matuda@gmail.com>

### CODE REVIEW

File: controllers/mysqlcluster_controller.go [hunk 1]
```diff
@@ -739,7 +740,8 @@ func (r *MySQLClusterReconciler) makePodTemplate(log logr.Logger, cluster *mocov
 			return nil, err
 		}
 	}
-	newTemplate.Spec.Containers = append(newTemplate.Spec.Containers, corev1.Container{
+	rooPasswordSecretName := rootPasswordSecretPrefix + moco.UniqueName(cluster)
```
Comment (kfyharukz) [inline controllers/mysqlcluster_controller.go:743]:
```suggestion
	rootPasswordSecretName := rootPasswordSecretPrefix + moco.UniqueName(cluster)
```

File: initialize/init_test.go [hunk 2]
```diff
@@ -251,6 +262,97 @@ func testTouchInitOnceCompleted(t *testing.T) {
...
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Second * 3)
+	}
+	if err != nil {
+		t.Fatal(err)
+	}
+	initPassword := "password"
+	_, err = db.Exec("CREATE USER IF NOT EXISTS 'init'@'localhost' IDENTIFIED BY ?", initPassword)
```
Comment (kfyharukz) [inline initialize/init_test.go:288]: user name `init` should be a variable

File: initialize/init_test.go [hunk 3]
```diff
@@ -251,6 +262,97 @@ func testTouchInitOnceCompleted(t *testing.T) {
...
+		moco.MiscUser,
+	} {
+		sqlRows, err := db.Query("SELECT user FROM mysql.user WHERE (user = ? AND host = '%')", k)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !sqlRows.Next() {
+			t.Errorf("user '%s' should be created, but not exist", k)
+		}
+	}
```
Comment (kfyharukz) [inline initialize/init_test.go:353]: `db.Close()`

File: operators/suite_test.go [hunk 4]
```diff
@@ -84,12 +81,13 @@ var _ = BeforeSuite(func(done Done) {
 	Expect(err).ToNot(HaveOccurred())
 	Expect(k8sClient).ToNot(BeNil())
 
-	stopMySQLD(mysqldName1)
-	stopMySQLD(mysqldName2)
-	removeNetwork()
+	moco.StopAndRemoveMySQLD(mysqldName1)
+	moco.StopAndRemoveMySQLD(mysqldName2)
```
Comment (kfyharukz) [inline operators/suite_test.go:85]: also remove `mysqldName`

File: operators/suite_test.go [hunk 5]
```diff
@@ -99,9 +97,9 @@ var _ = AfterSuite(func() {
 	err := testEnv.Stop()
 	Expect(err).ToNot(HaveOccurred())
 
-	stopMySQLD(mysqldName1)
-	stopMySQLD(mysqldName2)
-	removeNetwork()
+	moco.StopAndRemoveMySQLD(mysqldName1)
+	moco.StopAndRemoveMySQLD(mysqldName2)
```
Comment (kfyharukz) [inline operators/suite_test.go:101]: ditto

File: operators/update_primary_test.go [hunk 6]
```diff
@@ -17,14 +20,18 @@ var _ = Describe("Update primary", func() {
 	ctx := context.Background()
 
 	BeforeEach(func() {
-		err := startMySQLD(mysqldName1, mysqldPort1, mysqldServerID1)
+		err := moco.StartMySQLD(mysqldName1, mysqldPort1, mysqldServerID1)
```
Comment (kfyharukz) [inline operators/update_primary_test.go:23]: should write func to initialize `mysqld`s

File: operators/constants.go [hunk 7]
```diff
@@ -7,6 +7,8 @@ const (
 	OperatorSetCloneDonorList = "set-clone-donor-list"
 	// OperatorClone is a name of the clone primary instance operation
 	OperatorClone = "clone"
+	// OperatorExternalClone is a name of the clone external primary instance operation
+	OperatorExternalClone = "external-clone"
```
Comment (umezawatakeshi) [inline operators/constants.go:11]: not used from anywhere

File: agent/health_test.go [hunk 8]
```diff
@@ -12,19 +12,55 @@ import (
...
-			ConnectionTimeout: 3 * time.Second,
-			ReadTimeout:       30 * time.Second,
-		},
-	)
+var _ = Describe("Test Agent: Health Request", func() {
+	var agent *Agent
+	var registry *prometheus.Registry
+
+	BeforeEach(func() {
+		err := moco.StartMySQLD(donorHost, donorPort, donorServerID)
```
[the remark this answers is not in the document]
Comment (kfyharukz) [reply agent/health_test.go:27]: do in a refactoring task

File: operators/update_primary_test.go (hunk 6 above)
Comment (kfyharukz) [reply operators/update_primary_test.go:23]: will do in a refactoring task

**Review by umezawatakeshi — APPROVED** [review]

**Pull request merged**